tests/Commands/ToggleDone.test.ts

```diff
@@ -106,9 +106,9 @@ describe('ToggleDone', () => {
 
         GlobalFilter.set('#task');
 
-        testToggleLine('|- ', '- [ ] |');
-        testToggleLine('- |', '- [ ] |');
-        testToggleLine('- |foobar', '- [ ] foobar|');
+        testToggleLine('|- ', '- [ ] #task |');
+        testToggleLine('- |', '- [ ] #task |');
+        testToggleLine('- |foobar', '- [ ] #task foobar|');
     });
```

Comment:
It would be good to see some more thorough testing here.

Such as, what happens when there is already `#task` present on the line. To ensure that it doesn't add a duplicate tag.

Also, can we test the behaviour with non-tag global filter please?

Also, what happens if the global filter contains characters which are special characters in regular expressions?
(There is some code somewhere to deal with that - like, escaping regex special characters in arbitrary strings).

That's all I can think of for now!

Reply:
Also it would be good to have tests for the case where there is no global filter defined, for example to guard against accidental insertion of a space at the beginning of the updated lines.

Reply:
Okay I've added more testing in 86b4372a3611fc223db94b979241e9fb2ad68181. Does that seem sufficient to you? Are there any other tests that are missing?

Some of those tests revealed a smaller bug which is also fixed in that commit. If the task description was empty then toggling the task done would insert an extra space.

`- |#task` would become `- [ ]  #task|` instead of `- [ ] #task|`

I also added handling for when the description already includes the global filter.

> Also, what happens if the global filter contains characters which are special characters in regular expressions?
I added some tests for this in 6d116e62971c343d8f8937e19b4c0d4b30ef06c5